src/core/infer_request.cc

```diff
@@ -33,6 +33,127 @@
 
 namespace nvidia { namespace inferenceserver {
 
+namespace {
+
+class PersistentBufferManager {
```

Comment:
This wasn't exactly what I had in mind. We still have a mutex on every alloc... I think you should just put this to the side for now and we can revisit once we have a better undertanding of the full flow.